### DISCUSSION TITLE
Chrome Dev Tools + Developing in VSCode

### DISCUSSION THREAD

**Discussion by jakobtroidl** [manzt/anywidget#482] [General] [answered]
I am trying to figure out my development workflow with _anywidget_ in VSCode. Is there any way to access Chrome developer tools or inspect the javascript console while testing my _*.ipynb_ in a VSCode environment? So far, I have only been able to do that when working in the jupyter lab. Overall, I find the VSCode developer experience much better than Jupyter Lab, and how to streamline my dev workflow.

**Comment by manzt** [comment, marked as answer]
Hey! Thanks for trying out anywidget. Yes, you should be about to open the Chrome developer tools via the command palette (Command + Shift + P for >, then search for "Developer: Toggle Developer Tools").

I usually recommend starting with the minimal JS template for VS Code development: `npm create anywidget@latest`. This will setup a few things and allow creating Python envs/etc with the VS Code UI (which should be a nice way to get started).


![Export-1710803978782](https://github.com/manzt/anywidget/assets/24403730/9ca4b9f7-fd9a-4a2d-a27c-fdfd53a27fa7)

**Reply by manzt** [reply]
Of course, you can also just open an `.ipynb` notebook and get started! Usually nice to have the Python code in the notebook initially (so you don't need to restart the kernel to see Python changes).

**Comment by jakobtroidl** [comment]
Amazing! That's very cool! I'll give that a try. Thanks for the swift response, @manzt :)
